CalibTracker/SiStripESProducers/plugins/real/SiStripGainSimESProducer.cc

```diff
@@ -0,0 +1,76 @@
+// system include files
+#include <memory>
+#include <utility>
+
+// user include files
+#include "FWCore/Framework/interface/ModuleFactory.h"
+#include "FWCore/Framework/interface/ESProducer.h"
+
+#include "FWCore/ParameterSet/interface/ParameterSet.h"
+#include "FWCore/MessageLogger/interface/MessageLogger.h"
+
+#include "CalibFormats/SiStripObjects/interface/SiStripGain.h"
+#include "CondFormats/SiStripObjects/interface/SiStripApvGain.h"
+#include "CondFormats/DataRecord/interface/SiStripCondDataRecords.h"
+#include "CalibTracker/Records/interface/SiStripDependentRecords.h"
+
+#include "SiStripGainFactor.h"
+
+class SiStripGainSimESProducer : public edm::ESProducer {
+ public:
+  SiStripGainSimESProducer(const edm::ParameterSet&);
+  ~SiStripGainSimESProducer() override{};
+  
+  std::unique_ptr<SiStripGain> produce(const SiStripGainSimRcd&);
+
+ private:
+  struct TokenLabel {
+    TokenLabel(edm::ESConsumesCollector& cc, std::string record, std::string label):
+      token_{cc.consumesFrom<SiStripApvGain, SiStripApvGainSimRcd>(edm::ESInputTag{"", label})},
+      recordLabel_{std::move(record), std::move(label)}
+    {}
+    edm::ESGetToken<SiStripApvGain, SiStripApvGainSimRcd> token_;
+    std::pair<std::string, std::string> recordLabel_;
+  };
+
+  std::vector<TokenLabel> tokenLabels_;
+  SiStripGainFactor factor_;
+};
+
+SiStripGainSimESProducer::SiStripGainSimESProducer(const edm::ParameterSet& iConfig):
+  factor_{iConfig}
+{
+  auto cc = setWhatProduced(this);
+
+  auto apvGainLabels = iConfig.getParameter<std::vector<edm::ParameterSet> >("APVGain");
+  if(apvGainLabels.empty()) {
+    throw cms::Exception("Configuration") << "Got empty APVGain vector, but need at least one entry";
+  }
+
+  // Fill the vector of apv labels
+  for(const auto& gainPSet: apvGainLabels) {
+    // Shouldn't all these parameters be tracked?
+    tokenLabels_.emplace_back(cc,  gainPSet.getParameter<std::string>("Record"), gainPSet.getUntrackedParameter<std::string>("Label", ""));
+    factor_.push_back_norm(gainPSet.getUntrackedParameter<double>("NormalizationFactor", 1.));
```

Comment:
Shouldn't all these parameters be tracked? (possible implications being beyond this PR though)

Reply:
If it is agreed that these parameters should be tracked, I'll open an issue.